xarray/core/indexing.py

```diff
@@ -303,7 +303,11 @@ class VectorizedIndexer(IndexerTuple):
     """ Tuple for vectorized indexing """
 
 
-class LazilyIndexedArray(utils.NDArrayMixin):
+class NDArrayIndexable(utils.NDArrayMixin):
```

Comment:
Rather than inheriting an implementation (which requires all the various changes to implementations above), why not simply make this a dummy mixin class instead?

We could still inherit from `NDArrayIndexable` to mark that a class supports indexing, but it would not add any methods, e.g.,
```python
class NDArrayIndexable(object):
    """Mixin to mark support for IndexerTuple subclasses in indexing."""
```